web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -1330,36 +1386,90 @@ class AppVersionHistory extends Component {
+          handleActionButtonClicked={() =>
+            this.handleActionButtonClicked({
+              sequence: version.sequence,
+              versionLabel: version.versionLabel,
+            })
+          }
+          isHelmManaged={this.props.isHelmManaged}
+          key={version.sequence}
+          app={this.props.app}
+          match={this.props.match}
+          history={this.props.history}
+          version={version}
+          selectedDiffReleases={this.state.selectedDiffReleases}
+          nothingToCommit={nothingToCommit}
+          isChecked={isChecked}
+          isNew={isNew}
+          newPreflightResults={newPreflightResults}
+          showReleaseNotes={this.showReleaseNotes}
+          renderDiff={this.renderDiff}
+          toggleShowDetailsModal={this.toggleShowDetailsModal}
+          gitopsEnabled={gitopsEnabled}
+          deployVersion={this.deployVersion}
+          redeployVersion={this.redeployVersion}
+          downloadVersion={this.downloadVersion}
+          upgradeAdminConsole={this.upgradeAdminConsole}
+          handleViewLogs={this.handleViewLogs}
+          handleSelectReleasesToDiff={this.handleSelectReleasesToDiff}
+          renderVersionDownloadStatus={this.renderVersionDownloadStatus}
+          isDownloading={
+            this.state.versionDownloadStatuses?.[version.sequence]
+              ?.downloadingVersion
+          }
+          adminConsoleMetadata={this.props.adminConsoleMetadata}
+        />
+        {this.state.showHelmDeployModalForVersionLabel ===
+          version.versionLabel && (
+          <UseDownloadValues
+            appSlug={this.props?.app?.slug}
+            fileName="values.yaml"
+          >
+            {({
+              download,
+              clearError: clearDownloadError,
+              error: downloadError,
+              isDownloading,
+              name,
+              ref,
+              url,
+            }) => {
+              return (
+                <>
+                  <HelmDeployModal
+                    appSlug={this.props?.app?.slug}
+                    chartPath={this.props?.app?.chartPath || ""}
+                    downloadClicked={download}
+                    error={downloadError}
+                    isDownloading={isDownloading}
+                    hideHelmDeployModal={() => {
+                      this.setState({ showHelmDeployModalForVersionLabel: "" });
+                      clearDownloadError();
+                    }}
+                    registryUsername={this.props?.app?.credentials?.username}
+                    registryPassword={this.props?.app?.credentials?.password}
+                    showHelmDeployModal={true}
+                    showDownloadValues={
+                      this.deployButtonStatus(version) !== "Redeploy"
+                    }
+                    subtitle="Follow the steps below to upgrade your application with your new values.yaml."
+                    title={` ${this.deployButtonStatus(version)} ${
```

Comment:
Would it make any difference for readability/ease of access (if we want to update it later on) to put this in a variable?

Reply:
I'm thinking of trying to move it into the react-query state using a selector.